sources/web/datalab/polymer/components/datalab-editor/datalab-editor.ts

```diff
@@ -116,6 +122,30 @@ class DatalabEditorElement extends Polymer.Element {
   }
 
   /**
+   * Saves the currently open file.
+   */
+  _saveAsync() {
+    // TODO: If the file isn't defined, this means it's a blank editor, we'll need
+    // to save a new file. Open a file picker dialog here to get the file path.
+    if (this._file) {
+      const filePath = this._file.path;
+      const dirPath = filePath.substr(0, filePath.lastIndexOf(this._file.name));
+      const model: JupyterFile = {
+        content: this._editor.doc.getValue(),
+        format: this._file.format,
+        mimetype: this._file.mimetype,
+        name: this._file.name,
+        path: dirPath,
+        type: this._file.type,
+      };
+      return ApiManager.saveJupyterFile(model);
+      // TODO: Handle save success/failure here.
+    } else {
+      return Promise.resolve(null);
```

Comment:
Should there be a dialog telling the user of the exp code that this is not yet implemented, or will you be adding that functionality right away?